packages/atomic/src/components/common/layout/sections.ts

```diff
@@ -7,20 +9,26 @@ export const resultSectionTags = [
   'atomic-result-section-emphasized',
   'atomic-result-section-excerpt',
   'atomic-result-section-bottom-metadata',
-] as const;
+] as const);
+
+type SetValueType<T> = T extends Set<infer U> ? U : never;
```

Comment:
this is just a quick helper I made, it's only needed here. I can extract it to a util file if you prefer.